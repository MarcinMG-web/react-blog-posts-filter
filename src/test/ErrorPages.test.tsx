import { screen } from '@testing-library/react';
import ErrorPages from '../pages/ErrorPages';
import { render } from './test-utils';

describe('Error pages', () => {
  test('should render correctly', () => {
    const { asFragment } = render(<ErrorPages errorCode={400} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders with 400 error code', () => {
    render(<ErrorPages errorCode={400} />);

    expect(screen.getByText('400')).toBeInTheDocument();
    expect(screen.getByText('Bad Request')).toBeInTheDocument();
    expect(screen.getByText('The server could not understand the request due to invalid syntax.')).toBeInTheDocument();
  });

  test('renders with 403 error code', () => {
    render(<ErrorPages errorCode={403} />);

    expect(screen.getByText('403')).toBeInTheDocument();
    expect(screen.getByText('Forbidden')).toBeInTheDocument();
    expect(screen.getByText('Sorry, you are not authorized to access this page.')).toBeInTheDocument();
  });

  test('renders with 404 error code', () => {
    render(<ErrorPages errorCode={404} />);

    expect(screen.getByText('404')).toBeInTheDocument();
    expect(screen.getByText('Not Found')).toBeInTheDocument();
    expect(screen.getByText('Sorry, the page you visited does not exist.')).toBeInTheDocument();
  });

  test('renders with default error code (500)', () => {
    render(<ErrorPages errorCode={500} />);

    expect(screen.getByText('500')).toBeInTheDocument();
    expect(screen.getByText('Internal Server Error')).toBeInTheDocument();
    expect(screen.getByText('Sorry, something went wrong.')).toBeInTheDocument();
  });
});
