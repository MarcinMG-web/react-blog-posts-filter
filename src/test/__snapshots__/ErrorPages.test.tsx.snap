// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Error pages > should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-1va92lp-JoyStack-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeXl2 css-1bfyuvs-MuiSvgIcon-root"
      data-testid="ErrorOutlineIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2M12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8"
      />
    </svg>
    <h1
      class="MuiTypography-root MuiTypography-h1 css-1o1xzye-JoyTypography-root"
    >
      400
    </h1>
    <span
      class="MuiTypography-root MuiTypography-body-xs css-1u5qaj1-JoyTypography-root"
    >
      Bad Request
    </span>
    <p
      class="MuiTypography-root MuiTypography-body-sm css-n3pxpz-JoyTypography-root"
    >
      The server could not understand the request due to invalid syntax.
    </p>
    <button
      class="MuiButton-root MuiButton-variantSolid MuiButton-colorPrimary MuiButton-sizeMd css-1titjnu-JoyButton-root"
      type="button"
    >
      Back
    </button>
  </div>
</DocumentFragment>
`;
